runtime/src/main/java/org/corfudb/runtime/view/AbstractView.java

```diff
@@ -114,16 +114,17 @@ public Layout getCurrentLayout() {
                             + "invalidate view", we.getCorrectEpoch());
                 } else if (re instanceof NetworkException) {
                     log.warn("layoutHelper: System seems unavailable", re);
-
-                    if (--systemDownTriggerCounter <= 0) {
-                        runtime.systemDownHandler.run();
-                    }
                 } else {
                     throw re;
                 }
                 if (rethrowAllExceptions) {
                     throw new RuntimeException(re);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)